cloud/team/team.go

```diff
@@ -0,0 +1,708 @@
+	}
+
+	fmt.Println("\nPlease select the team you would like to update:")
+
+	teamMap := map[string]astrocore.Team{}
+	for i := range teams {
+		index := i + 1
+		table.AddRow([]string{
+			strconv.Itoa(index),
+			teams[i].Name,
+			teams[i].Id,
+		}, false)
+		teamMap[strconv.Itoa(index)] = teams[i]
+	}
+
+	table.Print(os.Stdout)
+	choice := input.Text("\n> ")
+	selected, ok := teamMap[choice]
+	if !ok {
+		return astrocore.Team{}, errInvalidTeamKey
+	}
+	return selected, nil
+}
+
+func GetWorkspaceTeams(client astrocore.CoreClient, workspace string, limit int) ([]astrocore.Team, error) {
+	offset := 0
+	var teams []astrocore.Team
+
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return nil, err
+	}
+	if ctx.OrganizationShortName == "" {
+		return nil, ErrNoShortName
+	}
+	if workspace == "" {
+		workspace = ctx.Workspace
+	}
+	for {
+		resp, err := client.ListWorkspaceTeamsWithResponse(httpContext.Background(), ctx.OrganizationShortName, workspace, &astrocore.ListWorkspaceTeamsParams{
+			Offset: &offset,
+			Limit:  &limit,
+		})
+		if err != nil {
+			return nil, err
+		}
+		err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		teams = append(teams, resp.JSON200.Teams...)
+
+		if resp.JSON200.TotalCount <= offset {
+			break
+		}
+
+		offset += limit
+	}
+
+	return teams, nil
+}
+
+// Prints a list of all of an organizations teams
+func ListWorkspaceTeams(out io.Writer, client astrocore.CoreClient, workspace string) error {
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if workspace == "" {
+		workspace = ctx.Workspace
+	}
+	table := printutil.Table{
+		Padding:        []int{10, 50, 50, 10, 10, 10, 10},
+		DynamicPadding: true,
+		Header:         []string{"ID", "Role", "Name", "Description", "CREATE DATE"},
+	}
+	teams, err := GetWorkspaceTeams(client, workspace, teamPagnationLimit)
+	if err != nil {
+		return err
+	}
+
+	for i := range teams {
+		var teamRole string
+		for _, role := range *teams[i].Roles {
+			if role.EntityType == "WORKSPACE" && role.EntityId == workspace {
+				teamRole = role.Role
+			}
+		}
+		table.AddRow([]string{
+			teams[i].Id,
+			teamRole,
+			teams[i].Name,
+			*teams[i].Description,
+			teams[i].CreatedAt.Format(time.RFC3339),
+		}, false)
+	}
+
+	table.Print(out)
+	return nil
+}
+
+func AddWorkspaceTeam(id, role, workspace string, out io.Writer, client astrocore.CoreClient) error {
+	err := user.IsWorkspaceRoleValid(role)
+	if err != nil {
+		return err
+	}
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return ErrNoShortName
+	}
+	if workspace == "" {
+		workspace = ctx.Workspace
+	}
+
+	// Get all org teams. Setting limit to 1000 for now
+	teams, err := GetOrgTeams(client)
+	if err != nil {
+		return err
+	}
+	if len(teams) == 0 {
+		return ErrNoTeamsFoundInOrg
+	}
+	teamID, err := getTeamID(id, teams)
```

Comment:
maybe we can reuse this code above in few places where we are iterating teams list to get the correct team out